modules/apm/src/main/java/module-info.java

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+module org.elasticsearch.tracing.apm {
+    requires org.elasticsearch.base;
+    requires org.elasticsearch.server;
```

Comment:
I'm totally not up-to-speed with such files, but I'm surprised that this is needed?

Reply:
It is in every other plugin because it's needed for things like `org.elasticsearch.env` and `org.elasticsearch.plugins`.

To test, I have individually removed each of these lines to see if the module still compiles.  They are all necessary.